cpp/src/lance/arrow/type.cc

```diff
@@ -75,8 +76,13 @@ ::arrow::Result<std::string> ToLogicalType(std::shared_ptr<::arrow::DataType> dt
                        dict_type->value_type()->ToString(),
                        dict_type->index_type()->ToString(),
                        dict_type->ordered());
+  } else if (::lance::arrow::is_extension(dtype)) {
+    // TODO what if extension_name has a `:` in it?
+    auto ext_type = std::dynamic_pointer_cast<::arrow::ExtensionType>(dtype);
```

Comment:
So if this is a struct type, how does it connect the sub-columns (i.e., {uri} column in `Image`) to the struct?

Reply:
this is just translating the logical type name right? We don’t need to connect the sub-columns until we’re actually reading and writing the schema metadata?